File: Components/Dashboard/VB/ASPxDashboard/App_Code/Data/ChampionsLeagueStatistics.vb
Imports Microsoft.VisualBasic
Imports System.Collections.Generic
Imports System.Data

Namespace DashboardMainDemo
	Public NotInheritable Class ChampionsLeagueStatisticsDataHelper
		Private Sub New()
		End Sub
		Public Shared Function Generate(ByVal dataSet As DataSet) As IList(Of ChampionsLeagueStatisticsDataRow)
			Dim res As IList(Of ChampionsLeagueStatisticsDataRow) = New List(Of ChampionsLeagueStatisticsDataRow)()
			Dim data As DataRowCollection = dataSet.Tables("Data").Rows
			For Each row As DataRow In data
				res.Add(New ChampionsLeagueStatisticsDataRow With {.Country = CStr(row("Country")), .Club = CStr(row("Club")), .Season = CStr(row("Season")), .Won = CInt(Fix(row("Won"))), .Drawn = CInt(Fix(row("Drawn"))), .Lost = CInt(Fix(row("Lost"))), .Goals_for = CInt(Fix(row("Goals_for"))), .Goals_against = CInt(Fix(row("Goals_against")))})
			Next row
			Return res
		End Function
	End Class

	Public Class ChampionsLeagueStatisticsDataRow
		Public Property Country() As String
		Public Property Club() As String
		Public Property Season() As String
		Public Property Won() As Integer
		Public Property Drawn() As Integer
		Public Property Lost() As Integer
		Public Property Goals_for() As Integer
		Public Property Goals_against() As Integer
	End Class
End Namespace

File: Components/Dashboard/CS/MVCxDashboard/Code/Data/ChampionsLeagueStatistics.cs
using System.Collections.Generic;
using System.Data;

namespace DashboardMainDemo {
    public static class ChampionsLeagueStatisticsDataHelper {
        public static IList<ChampionsLeagueStatisticsDataRow> Generate(DataSet dataSet) {
            IList<ChampionsLeagueStatisticsDataRow> res = new List<ChampionsLeagueStatisticsDataRow>();
            DataRowCollection data = dataSet.Tables["Data"].Rows;
            foreach(DataRow row in data) {
                res.Add(new ChampionsLeagueStatisticsDataRow {
                    Country = (string)row["Country"],
                    Club = (string)row["Club"],
                    Season = (string)row["Season"],
                    Won = (int)row["Won"],
                    Drawn = (int)row["Drawn"],
                    Lost = (int)row["Lost"],
                    Goals_for = (int)row["Goals_for"],
                    Goals_against = (int)row["Goals_against"]
                });
            }
            return res;
        }
    }

    public class ChampionsLeagueStatisticsDataRow {
        public string Country { get; set; }
        public string Club { get; set; }
        public string Season { get; set; }
        public int Won { get; set; }
        public int Drawn { get; set; }
        public int Lost { get; set; }
        public int Goals_for { get; set; }
        public int Goals_against { get; set; }
    }
}
